frontend/src/citizen-frontend/calendar/DayView.tsx

```diff
@@ -40,53 +60,256 @@ export default React.memo(function DayView({ date, data, selectDate }: Props) {
-            <Label>Varaus</Label>
-            {reservation === undefined ? (
-              <NoReservation>Ei varausta</NoReservation>
-            ) : (
-              <span>{`${formatDate(
-                reservation.startTime,
-                DATE_FORMAT_TIME_ONLY
-              )} – ${formatDate(
-                reservation.endTime,
-                DATE_FORMAT_TIME_ONLY
-              )}`}</span>
-            )}
-            <Label>Toteuma</Label>
-            <span>–</span>
-          </Grid>
-        </Fragment>
-      ))}
+      {editing
+        ? editorState.map(({ child, startTime, endTime, errors }, index) => (
+            <Fragment key={child.id}>
+              {index !== 0 ? <Separator /> : null}
+              <H3 noMargin>
+                {child.preferredName || child.firstName.split(' ')[0]}
+              </H3>
+              <Gap size="s" />
+              <Grid>
+                <Label>Varaus</Label>
+                <InputWrapper>
+                  <InputField
+                    type="time"
+                    onChange={editorStateSetter(index, 'startTime')}
+                    value={startTime}
+                    info={errorToInputInfo(
+                      errors.startTime,
+                      i18n.validationErrors
+                    )}
+                    readonly={saving}
+                  />
+                  –
+                  <InputField
+                    type="time"
+                    onChange={editorStateSetter(index, 'endTime')}
+                    value={endTime}
+                    info={errorToInputInfo(
+                      errors.endTime,
+                      i18n.validationErrors
+                    )}
+                    readonly={saving}
+                  />
+                </InputWrapper>
+                <Label>Toteuma</Label>
+                <span>–</span>
+              </Grid>
+            </Fragment>
+          ))
+        : childrenWithReservations.map(({ child, reservation }, index) => (
+            <Fragment key={child.id}>
+              {index !== 0 ? <Separator /> : null}
+              <H3 noMargin>
+                {child.preferredName || child.firstName.split(' ')[0]}
+              </H3>
+              <Gap size="s" />
+              <Grid>
+                <Label>Varaus</Label>
+                {reservation === undefined ? (
+                  <NoReservation>Ei varausta</NoReservation>
+                ) : (
+                  <span>{`${formatDate(
+                    reservation.startTime,
+                    DATE_FORMAT_TIME_ONLY
+                  )} – ${formatDate(
+                    reservation.endTime,
+                    DATE_FORMAT_TIME_ONLY
+                  )}`}</span>
+                )}
+                <Label>Toteuma</Label>
+                <span>–</span>
+              </Grid>
+            </Fragment>
+          ))}
+      {confirmationModal ? (
+        <InfoModal
+          title="Haluatko tallentaa muutokset?"
```

Comment:
Shouldn't these be localized? These will after all be read by screenreaders.

Reply:
Yeah, everything should be localized when it's enabled in production. And this is an actual "pop up" modal so it's definitely visible to everyone.